src/transformers/visitors/toCircuitVisitor.ts

```diff
@@ -164,12 +164,48 @@ const visitor = {
   },
 
   ParameterList: {
-    enter(path: NodePath) {
-      const { node, parent } = path;
+    enter(path: NodePath, state: any) {
+      const { node, parent, scope } = path;
+      let returnName : string;
+       if(path.key === 'parameters'){
       const newNode = buildNode('ParameterList');
       node._newASTPointer = newNode.parameters;
       parent._newASTPointer[path.containerName] = newNode;
-    },
+    } else if(path.key === 'returnParameters'){
+       parent.body.statements.forEach(node => {
+        if(node.nodeType === 'Return'){
+
+          // console.log(referencedId);
+          for(const [ id , bindings ] of Object.entries(scope.referencedBindings)){
+            if(id == node.expression.referencedDeclaration) {
+              if ((bindings instanceof VariableBinding))
+            state.returnIsSecret =bindings.isSecret
+          }
+          }
+          returnName = node.expression.name;
+          if(!returnName)
+          returnName = node.expression.value;
+        }
+      });
+
+    node.parameters.forEach(node => {
+    if(node.nodeType === 'VariableDeclaration'){
+    node.name = returnName;
```

Comment:
quick question on this - wouldn't this just rename every return param to `returnName`? So if we had more than one return param they would be overwritten?